gobblin-metrics-libs/gobblin-metrics/src/main/java/org/apache/gobblin/metrics/OpenTelemetryMetrics.java

```diff
@@ -54,6 +57,10 @@ private OpenTelemetryMetrics(State state) {
   protected MetricExporter initializeMetricExporter(State state) {
     OtlpHttpMetricExporterBuilder httpExporterBuilder = OtlpHttpMetricExporter.builder();
     httpExporterBuilder.setEndpoint(state.getProp(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_ENDPOINT));
+    if (state.contains(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_HEADER_KEY)) {
+      httpExporterBuilder.addHeader(state.getProp(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_HEADER_KEY),
+          state.getProp(ConfigurationKeys.METRICS_REPORTING_OPENTELEMETRY_HEADER_VALUE));
```

Comment:
I enforced both header key and value to exist and put a precondition on the reporting URL to avoid this

Reply:
good call.  thinking defaults here: is there a canonical value OT would want or it's deployment-specific, such as perhaps an endpoint URL?

Reply:
Sometimes, at least for microsoft geneva it can be pretty specific (attach account name and namespace) but this also depends greatly depending if you are emitting to an endpoint or a metric processor like fluentbit